oidc/viewsets.py

```diff
@@ -302,8 +331,11 @@ def callback(self, request: HttpRequest, **kwargs: dict) -> HttpResponse:  # noq
                     decoded_token = client.verify_and_decode_id_token(id_token)
                     if decoded_token.get(REDIRECT_AFTER_AUTH):
                         redirect_after = decoded_token.pop(REDIRECT_AFTER_AUTH)
-                    user_data.update(decoded_token)
-                    user_data = self.map_claims_to_model_field(user_data)
+                    user_data = self.map_claims_to_model_field(decoded_token)
+                    # Custom username provided by the user in case
+                    # a user with the same preferred username already exists
+                    form_data = request.POST.dict()
```

Comment:
@kelvin-muchiri, now that the endpoint supports both GET and POST, will this only be reached by POST requests?